sdk/core/core-http/.eslintrc.json

```diff
@@ -1,7 +1,11 @@
 {
   "plugins": ["@azure/azure-sdk"],
-  "extends": ["../../.eslintrc.old.json", "plugin:@azure/azure-sdk/recommended"],
-  "parserOptions": {
-    "createDefaultProgram": true
+  "extends": ["plugin:@azure/azure-sdk/azure-sdk-base"],
+  "rules": {
+    "@typescript-eslint/no-unused-vars": "off",
+    "@azure/azure-sdk/ts-use-interface-parameters": "off",
+    "@azure/azure-sdk/ts-package-json-main-is-cjs": "off",
```

Comment:
We can enable these rules, but we would need to upgrade all packages depending on this one.

Reply:
I'm not sure that I understand this comment? We should be able to avoid interface params or at least disable those ones selectively.

Also curious what ts-package-json-main-is-cjs actually does.

Reply:
These ts-package-json ones check for package.json properties and even fix them automatically, but they cause other projects to fail. Do I apply these fixes and make sure things build, or do I keep these properties as they are on master?

Reply:
I'm guessing the readme here should help, but the links seem to be broken: https://github.com/Azure/azure-sdk-for-js/tree/master/common/tools/eslint-plugin-azure-sdk

Reply:
that rule wants the package's `main` entry to be set to `"dist/index.js"`

Reply:
I pushed a change that brings back this rule.